protobuf/src/main/java/io/grpc/protobuf/ProtoUtils.java

```diff
@@ -107,7 +107,8 @@ private T parseFrom(InputStream stream) throws InvalidProtocolBufferException {
    */
   public static <T extends Message> Metadata.Key<T> keyForProto(final T instance) {
     return Metadata.Key.of(
-        instance.getDescriptorForType().getFullName() + Metadata.BINARY_HEADER_SUFFIX,
+        instance.getDescriptorForType().getFullName().replace('.', '-')
```

Comment:
The protobuf generator (or ~some~ generator is using it, at least internally).  

We can either:
  a.  roll this back
  b.  make a special case for this 
  c.  add period to the header name allowed chars.